src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1720,20 +1720,48 @@ def update_load_balancer_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         :return: the ManagedCluster object
         """
+        existing_managed_outbound_ips = None
+        if (
+            mc.network_profile and
+            mc.network_profile.load_balancer_profile and
+            mc.network_profile.load_balancer_profile.managed_outbound_i_ps
+        ):
+            existing_managed_outbound_ips = mc.network_profile.load_balancer_profile.managed_outbound_i_ps
+
         mc = super().update_load_balancer_profile(mc)
-        lb_profile = mc.network_profile.load_balancer_profile
 
-        if self.context.get_load_balancer_managed_outbound_ipv6_count() is not None:
-            lb_profile = update_load_balancer_profile(
-                self.context.get_load_balancer_managed_outbound_ip_count(),
-                self.context.get_load_balancer_managed_outbound_ipv6_count(),
-                self.context.get_load_balancer_outbound_ips(),
-                self.context.get_load_balancer_outbound_ip_prefixes(),
-                self.context.get_load_balancer_outbound_ports(),
-                self.context.get_load_balancer_idle_timeout(),
-                lb_profile,
-            )
-        mc.network_profile.load_balancer_profile = lb_profile
+        lb_managed_outbound_ipv6_count = self.context.get_load_balancer_managed_outbound_ipv6_count()
+        lb_managed_outbound_ip_count = self.context.get_load_balancer_managed_outbound_ip_count()
+        lb_outbound_ips = self.context.get_load_balancer_outbound_ips()
+        lb_outbound_ip_prefixes = self.context.get_load_balancer_outbound_ip_prefixes()
+        lb_outbound_ports = self.context.get_load_balancer_outbound_ports()
+        lb_idle_timeout = self.context.get_load_balancer_idle_timeout()
+
+        if (
+            not lb_managed_outbound_ipv6_count and
+            not lb_outbound_ips and
+            not lb_outbound_ip_prefixes and
+            existing_managed_outbound_ips
+        ):
+            lb_managed_outbound_ipv6_count = existing_managed_outbound_ips.count_ipv6
+
+        if (
+            not lb_managed_outbound_ip_count and
+            not lb_outbound_ips and
+            not lb_outbound_ip_prefixes and
+            existing_managed_outbound_ips
+        ):
+            lb_managed_outbound_ip_count = existing_managed_outbound_ips.count
```

Comment:
I suppose we may also combine these two ifs
```suggestion
        if (
            not lb_outbound_ips and
            not lb_outbound_ip_prefixes and
            existing_managed_outbound_ips
        ):
            if not lb_managed_outbound_ip_count:
                lb_managed_outbound_ip_count = existing_managed_outbound_ips.count
            if not lb_managed_outbound_ipv6_count:
                lb_managed_outbound_ipv6_count = existing_managed_outbound_ips.count_ipv6
```